### ISSUE TITLE
Feature Request: detect invalid/unset variable names in unset 

### ISSUE THREAD

**Issue by austin987** [koalaman/shellcheck#2466]
#### For new checks and feature suggestions
- [X] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [X] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/sh

foo_bar=blah
echo "${foo_bar}"
unset foo-bar
echo hi
```

#### Here's what shellcheck currently says:

Nothing

#### Here's what I wanted or expected to see:

`SCXXXX: foo-bar is an invalid variable name` (contains a hyphen, not underscore)

Note: this syntax error will throw an error on dash (bash happily ignores):
```sh

$ dash foo.sh 
blah
foo.sh: 5: unset: foo-bar: bad variable name

$ bash foo.sh 
blah
hi
```
